src/app/shared/modules/sai-editor/sai-editor.component.integration.spec.ts

```diff
@@ -542,13 +546,22 @@ describe('SaiEditorComponent integration tests', () => {
       expect(page.target2Name.innerText).toContain('param2');
       expect(page.target3Name.innerText).toContain('param3');
       expect(page.target4Name.innerText).toContain('param4');
+      expect(page.targetXName.innerText).toContain('TargetX');
+      expect(page.targetYName.innerText).toContain('TargetY');
+      expect(page.targetZName.innerText).toContain('TargetZ');
+      expect(page.targetOName.innerText).toContain('TargetO');
 
       page.setSelectValueById('target_type', 9);
 
       expect(page.target1Name.innerText).toContain('CreatureId');
       expect(page.target2Name.innerText).toContain('MinDistance');
       expect(page.target3Name.innerText).toContain('MaxDistance');
       expect(page.target4Name.innerText).toContain('AliveState');
+
+      page.setInputValueById('target_type', '28: 201'); // need this hack because option values are no longer in sequence at this point
+      expect(page.targetOName.innerText).toContain('Resize');
+
+      // TODO: add tests for target X, Y, Z dynamic names when they will be in use
```

Comment:
@Kitzunu 5) we add as many automated tests as possible, so new functionality = new tests :)